### PR TITLE
Sdk osgi support

### PR DESCRIPTION
**Pull request by acbeni**
@katmatt can you please review my code,

PS:The build is still failing due to the queue problem but locally they run fine

### CODE REVIEW

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/models/LocalizedString.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
 
     private static final Comparator<Map.Entry<Locale, String>> BY_LOCALE_COMPARATOR = (left, right) -> left.getKey().toString().compareTo(right.getKey().toString());
 
+    public static String getJsonCreatorName(){
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/models/LocalizedString.java]: My IDE tells me that we don't need this method. Can we remove it or is it used via reflection?

File: osgi-support/commercetools-tests-fragment/pom.xml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        <dependency>
+            <groupId>com.commercetools.sdk.jvm.core</groupId>
+            <artifactId>commercetools-sdk-base</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+        </dependency>
+        <dependency>
+            <groupId>com.commercetools.sdk.jvm.core</groupId>
+            <artifactId>commercetools-models</artifactId>
+            <version>1.21.1-SNAPSHOT</version>
```
Comment (katmatt) [inline osgi-support/commercetools-tests-fragment/pom.xml]: Please use `${project.version}` here, otherwise our release process might break.

File: commercetools-internal-docs/pom.xml [hunk 3]
```diff
@@ -38,6 +43,11 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
+            <groupId>com.commercetools.sdk.jvm.core</groupId>
+            <artifactId>commercetools-sdk-base</artifactId>
+            <version>1.21.1-SNAPSHOT</version>
```
Comment (katmatt) [inline commercetools-internal-docs/pom.xml]: Please use `${project.version}` here.

File: osgi-support/sdk-osgi-test-campaign/pom.xml [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.commercetools.sdk.jvm.core</groupId>
+            <artifactId>commercetools-test-lib</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.commercetools.sdk.jvm.core</groupId>
+            <artifactId>commercetools-sdk-base</artifactId>
+            <version>1.21.1-SNAPSHOT</version>
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: Please use `${project.version}` here.

File: commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java [hunk 5]
```diff
@@ -134,22 +131,6 @@
  <li class=fixed-in-release></li>
  </ul>
  -->
- <h3 class=released-version id="v1_21_0">1.21.0 (18.07.2017)</h3>
```
Comment (katmatt) [inline commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java]: This looks like you accidentially removed our release notes, should be fixed by merging master to this branch.

File: commercetools-java-client-ahc-1_9/pom.xml [hunk 6]
```diff
@@ -43,6 +43,13 @@
         </dependency>
     </dependencies>
     <build>
+        <resources>
```
Comment (katmatt) [inline commercetools-java-client-ahc-1_9/pom.xml]: Is this really required?

File: commercetools-sdk-base/pom.xml [hunk 7]
```diff
@@ -22,17 +22,17 @@
             <artifactId>sdk-http</artifactId>
             <version>${project.version}</version>
         </dependency>
-
+        <dependency>
+            <groupId>javax.money</groupId>
+            <artifactId>money-api</artifactId>
+            <version>1.0.1</version>
```
Comment (katmatt) [inline commercetools-sdk-base/pom.xml]: Here we should introduce a new property `money-api.version` to our parent pom and reference it here with `${money-api.version}`.

File: commercetools-sdk-base/pom.xml [hunk 8]
```diff
@@ -22,17 +22,17 @@
...
         <dependency>
             <groupId>org.javamoney</groupId>
             <artifactId>moneta</artifactId>
-            <exclusions>
-                <exclusion>
-                    <groupId>javax.annotation</groupId>
-                    <artifactId>javax.annotation-api</artifactId>
-                </exclusion>
-            </exclusions>
+            <version>1.1</version>
```
Comment (katmatt) [inline commercetools-sdk-base/pom.xml]: Please replace this version with `${moneta.version}`.

File: commercetools-sdk-base/src/main/resources/META-INF/services/javax.money.spi.ServiceProvider [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2012, 2015, Credit Suisse (Anatole Tresch), Werner Keil and others by the @author tag.
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/resources/META-INF/services/javax.money.spi.ServiceProvider]: Do we need the copy right notice here? To me it seems unrelated to their code or am I wrong?

File: sdk-http/pom.xml [hunk 10]
```diff
@@ -26,6 +26,35 @@
...
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>${felix.maven.version}</version>
+                <extensions>true</extensions>
+                <configuration>
```
Comment (katmatt) [inline sdk-http/pom.xml]: Why don't we use `<niceManifest>true</niceManifest>` here?

File: commercetools-java-client-core/pom.xml [hunk 11]
```diff
@@ -28,6 +28,42 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>${felix.maven.version}</version>
+                <extensions>true</extensions>
+
+                <configuration>
+
+                    <instructions>
```
Comment (katmatt) [inline commercetools-java-client-core/pom.xml]: Why don't we use `<niceManifest>true</niceManifest>` here?

File: commercetools-java-client-apache-async/pom.xml [hunk 12]
```diff
@@ -40,6 +40,37 @@
...
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>${felix.maven.version}</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
```
Comment (katmatt) [inline commercetools-java-client-apache-async/pom.xml]: Why don't we use `<niceManifest>true</niceManifest>?

File: osgi-support/sdk-osgi-test-campaign/pom.xml [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.ops4j.pax.exam</groupId>
+            <artifactId>pax-exam-junit4</artifactId>
+            <version>${pax-exam.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.ops4j.pax.logging</groupId>
+            <artifactId>pax-logging-api</artifactId>
+            <version>1.10.1</version>
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: We should use `${pax-exam.version}` here too.

File: osgi-support/sdk-osgi-test-campaign/pom.xml [hunk 14]
```diff
@@ -0,0 +1,268 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.ops4j.pax.logging</groupId>
+            <artifactId>pax-logging-service</artifactId>
+            <version>1.10.1</version>
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: We should use `${pax-exam.version}` here too.

File: osgi-support/sdk-osgi-test-campaign/pom.xml [hunk 15]
```diff
@@ -0,0 +1,268 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.framework</artifactId>
+            <version>5.6.2</version>
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: Here we should use `${felix.version}`

File: osgi-support/sdk-osgi-test-campaign/pom.xml [hunk 16]
```diff
@@ -0,0 +1,268 @@
...
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${junit.version}</version>
+            <scope>compile</scope>
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: Do we have to use scope `compile` here? If we have to use `compile` here, we should add a comment to explain it a bit.

File: osgi-support/sdk-osgi-test-campaign/pom.xml [hunk 17]
```diff
@@ -0,0 +1,268 @@
...
+
+
+        <dependency>
+            <groupId>org.ops4j.pax.exam</groupId>
+            <artifactId>pax-exam-testforge</artifactId>
+            <version>3.6.0</version>
+            <scope>test</scope>
+        </dependency>
+
+        <!--Dependencies for ServiceLoader based providers-->
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: Great that you added a doc here 👍

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+        }
+        return runners;
+    }
+
+    private static ClassesFinder createFinder(final Class<?> suiteClass,final  ClassesFinderFactory finderFactory) {
+        return finderFactory.create(getSearchInJars(suiteClass), getClassnameFilters(suiteClass), getSuiteTypes(suiteClass),
+                getBaseTypes(suiteClass), getExcludedBaseTypes(suiteClass), getClasspathProperty(suiteClass));
+    }
+
+    private static Class<?>[] getSortedTestclasses(final ClassesFinder finder) {
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java]: May be rename the method to `getSortedTestClasses`?

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+        List<Class<?>> testclasses = finder.find();
+        Collections.sort(testclasses, getClassComparator());
+        return testclasses.toArray(new Class[testclasses.size()]);
+    }
+
+    private static Comparator<Class<?>> getClassComparator() {
+        return Comparator.comparing(Class::getName);
+    }
+
+    private static String[] getClassnameFilters(final Class<?> suiteClass) {
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java]: May be rename the method to 'getClassNameFilters`?

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.junit.extensions.cpsuite.ClasspathSuite.*;
+
+public class PaxExamClasspathSuite extends Suite {
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java]: Would be great if you could add some javadoc to this class ;-)

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/annotations/MinimumTestClassesInSuite.java [hunk 21]
```diff
@@ -0,0 +1,24 @@
+package io.sphere.sdk.test.annotations;
+
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/annotations/MinimumTestClassesInSuite.java]: Nice javadoc 👍

File: sdk-http-apache-async/src/main/java/io/sphere/sdk/http/ApacheHttpClientAdapterImpl.java [hunk 22]
```diff
@@ -97,7 +99,10 @@ private HttpUriRequest toApacheRequest(final HttpRequest httpRequest) throws Uns
             final HttpRequestBody body = httpRequest.getBody();
             final HttpEntity httpEntity;
             if (body instanceof StringHttpRequestBody) {
-                httpEntity = stringEntityOf(((StringHttpRequestBody) body).getString());
+                StringEntity stringEntity = new StringEntity(((StringHttpRequestBody) body).getString(),"UTF-8");
```
Comment (katmatt) [inline sdk-http-apache-async/src/main/java/io/sphere/sdk/http/ApacheHttpClientAdapterImpl.java]: We could make this vraible `final`. And instead of `UTF-8` we could use `java.nio.charset.StandardCharsets.UTF_8.name()`

**Comment by katmatt** [comment]
Just another remark: I mentioned that we should use `<niceManifest>true</niceManifest>` everywhere. but instead of repeating this everywhere we should add this configuration to our parent pom.

File: commercetools-sdk-base/pom.xml [hunk 23]
```diff
@@ -74,16 +80,132 @@
...
+                            osgi.serviceloader;
+                            osgi.serviceloader="javax.money.spi.MonetaryCurrenciesSingletonSpi",
+                            osgi.serviceloader;
+                            osgi.serviceloader="javax.money.spi.RoundingProviderSpi",
+                            osgi.serviceloader;
+                            osgi.serviceloader="javax.money.spi.ServiceProvider",
+                            osgi.serviceloader;
+                            osgi.serviceloader="org.javamoney.moneta.spi.LoaderService"
+                        </Provide-Capability>
+                        <Embed-Dependency>scope=compile|runtime;inline=!META-INF/services/javax.money.spi.ServiceProvider;artifactId=money-api|moneta</Embed-Dependency>
```
Comment (katmatt) [inline commercetools-sdk-base/pom.xml]: Embedding makes a lot of sense here, nice that it works now 👍

File: commercetools-java-client-apache-async/pom.xml [hunk 24]
```diff
@@ -40,6 +40,37 @@
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
```
Comment (katmatt) [inline commercetools-java-client-apache-async/pom.xml]: May be we could pull this configuration into our parent pom and we should check that we don't release our `test-jar`s to maven central.

File: commercetools-sdk-base/pom.xml [hunk 25]
```diff
@@ -74,16 +80,132 @@
             <groupId>com.fasterxml.jackson.module</groupId>
             <artifactId>jackson-module-parameter-names</artifactId>
             <version>${jackson.version}</version>
+            <exclusions>
```
Comment (katmatt) [inline commercetools-sdk-base/pom.xml]: Would be nice if we could add these exclusions to our parent pom, just to avoid potential duplication.

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/utils/OSGiPriorityAwareServiceProvider.java [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+import javax.money.spi.ServiceProvider;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public final class OSGiPriorityAwareServiceProvider implements ServiceProvider {
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/utils/OSGiPriorityAwareServiceProvider.java]: Would be nice to have some javdoc here 😉

File: osgi-support/mirror-subclass-plugin/src/main/java/io/sphere/sdk/plugin/ChildCreationMojo.java [hunk 27]
```diff
@@ -0,0 +1,144 @@
...
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+
+@Mojo(name = "process-classes", threadSafe = true, defaultPhase = LifecyclePhase.PROCESS_CLASSES)
+public class ChildCreationMojo extends AbstractMojo {
```
Comment (katmatt) [inline osgi-support/mirror-subclass-plugin/src/main/java/io/sphere/sdk/plugin/ChildCreationMojo.java]: Would be nice to have some javadoc here 😉

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/JvmSdkConfigurationFactory.java [hunk 28]
```diff
@@ -0,0 +1,111 @@
...
+import org.ops4j.pax.exam.options.UrlProvisionOption;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.ops4j.pax.exam.CoreOptions.*;
+
+
+public class JvmSdkConfigurationFactory implements ConfigurationFactory {
```
Comment (katmatt) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/JvmSdkConfigurationFactory.java]: Would be nice to have some javadoc here 😉

File: commercetools-java-client-ahc-2_0/pom.xml [hunk 29]
```diff
@@ -37,6 +37,13 @@
         </dependency>
     </dependencies>
     <build>
+        <resources>
```
Comment (acbeni) [inline commercetools-java-client-ahc-2_0/pom.xml]: yeah jus forgot it there 😄 , i will remove it

File: commercetools-models/src/test/java/io/sphere/sdk/errors/ExceptionTestDsl.java [hunk 30]
```diff
@@ -0,0 +1,33 @@
...
+
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.client.ErrorResponseException;
+import io.sphere.sdk.client.SphereRequest;
+import io.sphere.sdk.models.errors.SphereError;
+import org.junit.rules.ExpectedException;
+
+import java.util.function.Supplier;
+
+public class ExceptionTestDsl {
```
Comment (acbeni) [inline commercetools-models/src/test/java/io/sphere/sdk/errors/ExceptionTestDsl.java]: 👍  totally agreed

File: commercetools-models/src/test/java/io/sphere/sdk/models/AddressTest.java [hunk 31]
```diff
@@ -5,7 +5,7 @@
 
 import static org.assertj.core.api.Assertions.*;
 
-public final class AddressTest {
+public class AddressTest {
```
Comment (acbeni) [inline commercetools-models/src/test/java/io/sphere/sdk/models/AddressTest.java]: Actually the subclassing plugin, needs the tests to be not final in order to be able to create subclasses without compile error, that's the main reason why i made so many classes not final, and that's also the reason why at least one of their constructors need to be non private

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/models/LocalizedString.java (hunk 1 above)
Comment (acbeni) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/models/LocalizedString.java]: You are right actually i think it needs to be removed, i believe i just forgot to rollback that change.

File: osgi-support/commercetools-tests-fragment/pom.xml (hunk 2 above)
Comment (acbeni) [inline osgi-support/commercetools-tests-fragment/pom.xml]: 👍  good catch

File: osgi-support/sdk-osgi-test-campaign/pom.xml (hunk 13 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: actually `${pax-exam.version}` is at `4.11.0` while i need 1.10.1 for pax-logging-api, i added a new var `${pax-exam.logging.version}` for these dependencies

File: osgi-support/sdk-osgi-test-campaign/pom.xml (hunk 14 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: actually `${pax-exam.version}` is at `4.11.0` while i need 1.10.1 for pax-logging-service, i added a new var `${pax-exam.logging.version}` for these dependencies

File: osgi-support/sdk-osgi-test-campaign/pom.xml (hunk 16 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: there is no need for 😅 , guess i ll remove it

File: osgi-support/sdk-osgi-test-campaign/pom.xml (hunk 4 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/pom.xml]: 👍  done!

File: commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java (hunk 5 above)
Comment (acbeni) [inline commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java]: Ok reversed 👍

File: commercetools-java-client-ahc-1_9/pom.xml (hunk 6 above)
Comment (acbeni) [inline commercetools-java-client-ahc-1_9/pom.xml]: No 😅  
removing it

File: commercetools-sdk-base/pom.xml (hunk 8 above)
Comment (acbeni) [inline commercetools-sdk-base/pom.xml]: done 👍

File: commercetools-sdk-base/pom.xml (hunk 7 above)
Comment (acbeni) [inline commercetools-sdk-base/pom.xml]: 👍  done

File: commercetools-sdk-base/pom.xml (hunk 23 above)
Comment (acbeni) [inline commercetools-sdk-base/pom.xml]: yep its solved a lot of problems 😓

File: commercetools-sdk-base/src/main/resources/META-INF/services/javax.money.spi.ServiceProvider (hunk 9 above)
Comment (acbeni) [inline commercetools-sdk-base/src/main/resources/META-INF/services/javax.money.spi.ServiceProvider]: actually it should be removed 👍  it's a leftover from wraping javax money to including it in sdk-base

File: sdk-http/pom.xml (hunk 10 above)
Comment (acbeni) [inline sdk-http/pom.xml]: i used `niceManifest` only when i needed to inspect the manifest file, but maybe adding it everywhere is a good idea.

File: commercetools-java-client-core/pom.xml (hunk 11 above)
Comment (acbeni) [inline commercetools-java-client-core/pom.xml]: added 👍

File: commercetools-java-client-apache-async/pom.xml (hunk 12 above)
Comment (acbeni) [inline commercetools-java-client-apache-async/pom.xml]: sorry didn't understood what you meant.

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java (hunk 20 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java]: Yeah agreed 👍 
i will add it to explain the purpose of this class

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java (hunk 18 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/PaxExamClasspathSuite.java]: sorry but the name is the same 😅

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/annotations/MinimumTestClassesInSuite.java (hunk 21 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/annotations/MinimumTestClassesInSuite.java]: 😄  thx

File: sdk-http-apache-async/src/main/java/io/sphere/sdk/http/ApacheHttpClientAdapterImpl.java (hunk 22 above)
Comment (acbeni) [inline sdk-http-apache-async/src/main/java/io/sphere/sdk/http/ApacheHttpClientAdapterImpl.java]: didn't know about the charset thing 👍 
but its better because now the code is even safer

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/utils/OSGiPriorityAwareServiceProvider.java (hunk 26 above)
Comment (acbeni) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/utils/OSGiPriorityAwareServiceProvider.java]: 👍  agreed

File: osgi-support/mirror-subclass-plugin/src/main/java/io/sphere/sdk/plugin/ChildCreationMojo.java (hunk 27 above)
Comment (acbeni) [inline osgi-support/mirror-subclass-plugin/src/main/java/io/sphere/sdk/plugin/ChildCreationMojo.java]: 👍  agreed

File: osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/JvmSdkConfigurationFactory.java (hunk 28 above)
Comment (acbeni) [inline osgi-support/sdk-osgi-test-campaign/src/main/java/io/sphere/sdk/test/JvmSdkConfigurationFactory.java]: 👍  agreed

**Comment by acbeni** [comment]
@katmatt Not all the jars are bundled, so i dont think we should use `<niceManifest>true</niceManifest>` at a . parent POM

File: commercetools-java-client-apache-async/pom.xml (hunk 24 above)
Comment (acbeni) [inline commercetools-java-client-apache-async/pom.xml]: yeah i vebeen able to pull this conf to parent pom but don't know ho to prevent from sharing the test-jar to maven central.

File: sdk-http/pom.xml (hunk 10 above)
Comment (acbeni) [inline sdk-http/pom.xml]: 👍  now added at the root level
